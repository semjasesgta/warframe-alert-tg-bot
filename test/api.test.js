/* eslint-env jest */
const nock = require('nock');
const dateFns = require('date-fns');
const {fetchData, formatAlert, formatInvasion} = require('../src/api');
const mockData = require('./fixtures/apiResponse.json');

// update mock data timings
mockData.Alerts.forEach(a => {
  a.Expiry.$date.$numberLong = dateFns.addDays(Date.now(), 1).getTime();
});

// mock response from warframe API
nock('http://content.warframe.com')
  .get('/dynamic/worldState.php')
  .reply(200, mockData);

// mock time now
const mockNow = new Date(2018, 0, 1, 1, 1, 1, 1);

// mock alert data for formatting
const mockAlert = {
  rewards: ['Alertium'],
  end: dateFns.addMinutes(mockNow, 40),
  start: dateFns.addMinutes(mockNow, 5),
};

const mockInvasion = {
  attackReward: ['OrokinReactorBlueprint'],
  defenderReward: ['OrokinCatalystBlueprint'],
  count: 1,
  goal: 100,
};

describe('Warframe API handling', () => {
  test('should fetch invasions and alerts', async done => {
    const result = await fetchData();
    // remove date from data to keep snapshot persistent
    result.alerts[0].end = 'fixed';
    // compare to snapshot
    expect(result).toMatchSnapshot();
    done();
  });

  test('should format alert info', () => {
    const alertText = formatAlert({alert: mockAlert, now: mockNow});
    expect(alertText).toMatchSnapshot();
  });

  test('should format invasion info', () => {
    const invasionText = formatInvasion(mockInvasion);
    expect(invasionText).toMatchSnapshot();
  });
});
