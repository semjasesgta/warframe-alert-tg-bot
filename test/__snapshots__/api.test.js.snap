// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warframe API handling should fetch invasions, alerts and sentient outpost 1`] = `
Object {
  "alerts": Array [
    Object {
      "end": "fixed",
      "id": "5c1020a14c4d2f1f193cc4dd",
      "rewards": Array [
        Object {
          "count": 1,
          "name": "Alertium",
        },
      ],
      "start": 2018-12-11T20:44:09.689Z,
      "type": "MT_TERRITORY",
    },
    Object {
      "end": 2099-02-02T00:01:01.000Z,
      "id": "5c1027318d42022a4052d4fb",
      "rewards": Array [
        Object {
          "count": 3,
          "name": "Forma",
        },
      ],
      "start": 2018-12-11T21:12:44.480Z,
      "type": "MT_TERRITORY",
    },
    Object {
      "end": 2099-02-02T00:01:01.000Z,
      "id": "5c102a3d3ff802992f5a1f00",
      "rewards": Array [
        Object {
          "count": 1,
          "name": "FormaUmbra",
        },
      ],
      "start": 2018-12-11T21:25:11.596Z,
      "type": "MT_DEFENSE",
    },
  ],
  "fomorian": Object {
    "count": 0,
    "end": 2021-11-12T19:59:37.685Z,
    "goal": 1000000,
    "id": "618678cd103bd85d64a5df51",
    "reward": Array [
      "OrokinCatalyst",
    ],
    "start": 2021-11-08T18:46:45.000Z,
  },
  "invasions": Array [
    Object {
      "attackReward": Array [],
      "count": -1623,
      "defenderReward": Array [
        Object {
          "count": 3,
          "name": "OrokinReactorBlueprint",
        },
      ],
      "goal": 45000,
      "id": "5c100cb560f09a7108813c9e",
    },
  ],
  "sentientOutpost": Object {
    "detectedDate": "fixed",
    "id": "sentient_outpost_510",
    "name": "Gian Point",
  },
}
`;

exports[`Warframe API handling should format alert info 1`] = `
"NEW ALERT:
Rewards: Nitain x2
Ends in: 5 days
Starts in: 5 minutes"
`;

exports[`Warframe API handling should format fomorian info 1`] = `
"NEW FOMORIAN EVENT:
Rewards: 
Ends in: 5 days
"
`;

exports[`Warframe API handling should format invasion info 1`] = `
"NEW INVASION: 
Rewards: Orokin Reactor Blueprint x2, Orokin Catalyst Blueprint
Current progress: ~1%"
`;

exports[`Warframe API handling should get and format baro info 1`] = `
"Baro is here (leaves in 23h)
Here's what he has:
   Prisma Grinlok Weapon - 500 ducats, 220000 credits
   Weapon Toxin Damage Mod Expert - 350 ducats, 200000 credits
   Elect Event Pistol Mod - 300 ducats, 150000 credits
   Elect Event Melee Mod - 300 ducats, 150000 credits
   Elect Event Rifle Mod - 300 ducats, 150000 credits
   Elect Event Shotgun Mod - 300 ducats, 150000 credits
   Prisma Obex - 500 ducats, 175000 credits
   Prisma Dual Cleavers - 490 ducats, 200000 credits
   Prisma Razor Scarf - 350 ducats, 275000 credits
   Baro Armour C - 150 ducats, 250000 credits
   Baro Armour A - 350 ducats, 110000 credits
   Baro Armour L - 300 ducats, 150000 credits
   Liset Insect Skin Prime Trader - 230 ducats, 375000 credits
   Baro Pet Mask - 500 ducats, 200000 credits
   Baro Pet Wings - 400 ducats, 250000 credits
   Baro Pet Tail - 400 ducats, 250000 credits
   Elixis Tigris - 300 ducats, 275000 credits
   Colour Picker Kiteer Item A - 150 ducats, 300000 credits
   Baro Melee Dangle - 250 ducats, 250000 credits
   Image Baro Kiteer - 80 ducats, 50000 credits
   Baro Mouth Piece A - 500 ducats, 400000 credits
   Baro Arrow - 375 ducats, 300000 credits
   Kubrow Pet Pattern Prime Trader A - 150 ducats, 300000 credits
   Mummy Quest Key Blueprint - 100 ducats, 25000 credits"
`;
